lib/stages/InputSubset.cpp

```diff
@@ -0,0 +1,197 @@
+#include "InputSubset.hpp"
+
+#include "Config.hpp"          // for Config
+#include "Hash.hpp"            // for operator<
+#include "StageFactory.hpp"    // for REGISTER_KOTEKAN_STAGE, StageMakerTemplate
+#include "buffer.h"            // for allocate_new_metadata_object, mark_frame_empty, mark_fram...
+#include "bufferContainer.hpp" // for bufferContainer
+#include "datasetManager.hpp"  // for dset_id_t, state_id_t, datasetManager
+#include "datasetState.hpp"    // for prodState
+#include "kotekanLogging.hpp"  // for FATAL_ERROR, WARN
+#include "visBuffer.hpp"       // for visFrameView, visField, visField::vis, visField::weight
+#include "visUtil.hpp"         // for prod_ctype, frameID, cmap, icmap, modulo, cfloat
+
+#include "gsl-lite.hpp" // for span
+
+#include <algorithm>    // for max, binary_search, copy, sort
+#include <atomic>       // for atomic_bool
+#include <complex>      // for complex
+#include <cxxabi.h>     // for __forced_unwind
+#include <exception>    // for exception
+#include <functional>   // for _Bind_helper<>::type, bind, function
+#include <future>       // for future, async
+#include <iterator>     // for back_insert_iterator, back_inserter
+#include <regex>        // for match_results<>::_Base_type
+#include <stdexcept>    // for out_of_range, runtime_error
+#include <stdint.h>     // for uint16_t, uint32_t
+#include <system_error> // for system_error
+#include <utility>      // for pair, tuple_element<>::type
+
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+using kotekan::Stage;
+
+REGISTER_KOTEKAN_STAGE(InputSubset);
+
+InputSubset::InputSubset(Config& config, const std::string& unique_name,
+                         bufferContainer& buffer_container) :
+    Stage(config, unique_name, buffer_container, std::bind(&InputSubset::main_thread, this)),
+    _inputs(config.get<std::set<uint32_t>>(unique_name, "inputs")) {
+
+    // Get buffers
+    in_buf = get_buffer("in_buf");
+    register_consumer(in_buf, unique_name.c_str());
+    out_buf = get_buffer("out_buf");
+    register_producer(out_buf, unique_name.c_str());
+}
+
+void InputSubset::change_dataset_state(dset_id_t ds_id) {
+
+    auto& dm = datasetManager::instance();
+
+    auto fprint = dm.fingerprint(ds_id, {"inputs", "products", "state"});
```

Comment:
I couldn't figure out what the `"state"` dataset is for.

Reply:
Ooops. Should be `stack`!

Reply:
Good catch. This line combined with the check in the branch below that there is no `stackState` is mean to guarantee you never get any stacked data coming through, this would definitely have broken it.